lib/grpc/adapter/cowboy.ex

```diff
@@ -153,11 +153,21 @@ defmodule GRPC.Adapter.Cowboy do
   end
 
   defp cowboy_start_args(endpoint, servers, port, opts) do
-    dispatch =
-      :cowboy_router.compile([
-        {:_, [{:_, GRPC.Adapter.Cowboy.Handler, {endpoint, servers, Enum.into(opts, %{})}}]}
-      ])
+    # Custom handler to be able to listen in the same port, more info: 
+    # https://github.com/containous/traefik/issues/6211
+    status_handler = Application.get_env(:grpc, :status_handler)
```

Comment:
I think we should avoid using global config as per the general library guidelines for Elixir. Instead, we should accept this through `opts`.

Is this able to be passed down already, or would something more need to change?

Reply:
Hi @polvalente, thanks for answering. This is something we still need. Let me try injecting the configuration using `opts` and check if it works.

Reply:
Yes! This is what I had in mind.
The benefit of doing it this way is that 2 processes can be started with different configs

Reply:
Great, much better. I've submitted the change!